data_steward/tools/snapshot_serology_dataset.py

```diff
@@ -0,0 +1,202 @@
+        LOGGER.info(f'Created table {snapshot_dataset_id}.{table}')
+
+
+def create_serology_snapshot(client: BigQueryClient, release_tag: str,
+                             src_serology_dataset_id: str) -> str:
+    """
+    Generates the serology snapshot dataset based on the release tag
+
+    :param client: a BigQueryClient
+    :param release_tag: Release tag for the CDR run
+    :param src_serology_dataset_id: Identifies the source serology dataset
+    :return: str: Identifies the created snapshot serology dataset
+    """
+    dataset_id = f"C{release_tag}_antibody_quest"
+    dataset = Dataset(f'{client.project}.{dataset_id}')
+    dataset.description = f'Source dataset: {src_serology_dataset_id} *_ct views; JIRA issue number: {ISSUE_NUMBER}'
+    dataset.labels = {'release_tag': release_tag, 'data_tier': 'controlled'}
+    dataset = client.create_dataset(dataset)
+    LOGGER.info(f'Successfully created empty dataset {dataset.dataset_id}')
+    return dataset.dataset_id
+
+
+def get_arg_parser():
+    parser = argparse.ArgumentParser(description="""
+    Creates a serology snapshot dataset as C{release_tag}_antibody_quest
+    using CT and existing serology dataset, R2020q4r1_antibody_quest.""")
+    parser.add_argument(
+        '-p',
+        '--project_id',
+        action='store',
+        dest='project_id',
+        help='Project associated with drc_ops and output dataset',
+        required=True)
+    parser.add_argument('-t',
+                        '--release_tag',
+                        action='store',
+                        dest='release_tag',
+                        help='Release tag for the CDR run',
+                        required=True)
+    parser.add_argument(
+        '-s',
+        '--src_serology_dataset_id',
+        action='store',
+        dest='src_serology_dataset_id',
+        help='Source serology dataset, currently R2020q4r1_antibody_quest',
+        required=True)
+    parser.add_argument('-c',
+                        '--ct_dataset_id',
+                        action='store',
+                        dest='ct_dataset_id',
+                        help='Controlled tier dataset',
+                        required=True)
+    parser.add_argument('-r',
+                        '--run_as_email',
+                        action='store',
+                        dest='run_as_email',
+                        help='Service account to impersonate',
+                        required=True)
+    return parser
+
+
+def main():
+    parser = get_arg_parser()
+    args = parser.parse_args()
+
+    # Set up pipeline logging
+    pipeline_logging.configure(add_console_handler=True)
+
+    # get credentials and create client
+    impersonation_creds = auth.get_impersonation_credentials(
+        args.run_as_email, CDR_SCOPES)
+
+    bq_client = BigQueryClient(args.project_id, credentials=impersonation_creds)
+
+    # Create serology dataset
+    dataset_id = create_serology_snapshot(bq_client, args.snapshot_date,
```

Comment:
args.snapshopt_date should be released tag.

Reply:
Thanks, updated it